Objects/moduleobject.c

```
@@ -679,12 +679,27 @@ module_repr(PyModuleObject *m)
 static PyObject*
 module_getattro(PyModuleObject *m, PyObject *name)
 {
-    PyObject *attr, *mod_name;
+    PyObject *attr, *mod_name, *getattr;
+    PyObject* stack[1] = {name};
     attr = PyObject_GenericGetAttr((PyObject *)m, name);
-    if (attr || !PyErr_ExceptionMatches(PyExc_AttributeError))
+    if (attr || !PyErr_ExceptionMatches(PyExc_AttributeError)) {
         return attr;
+    }
     PyErr_Clear();
     if (m->md_dict) {
+        _Py_IDENTIFIER(__getattr__);
+        getattr = _PyDict_GetItemId(m->md_dict, &PyId___getattr__);
+        if (getattr) {
+            if (!PyCallable_Check(getattr)) {
+                PyErr_SetString(PyExc_TypeError,
+                                "module __getattr__ must be callable");
+                return NULL;
+            }
+            return _PyObject_FastCall(getattr, stack, 1);
+        }
+        else if (PyErr_Occurred()) {
```

This never happen with `_PyDict_GetItemId()`.

I prefer to use `_PyDict_GetItemIdWithError()` and don't silence errors, but this is a different large issue.